cosmos/providers/databricks/plugin.py

```diff
@@ -0,0 +1,279 @@
+"""DatabricksWorkflowTaskGroup for submitting jobs to Databricks."""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    pass
+from operator import itemgetter
+
+from airflow.models import BaseOperator, BaseOperatorLink
+from airflow.models.dag import DAG
+from airflow.models.dagrun import DagRun
+from airflow.models.taskinstance import TaskInstance, TaskInstanceKey
+from airflow.models.xcom import XCom
+from airflow.plugins_manager import AirflowPlugin
+from airflow.providers.databricks.hooks.databricks import DatabricksHook
+from airflow.security import permissions
+from airflow.utils.airflow_flask_app import get_airflow_app
+from airflow.utils.session import provide_session
+from airflow.www.auth import has_access
+from airflow.www.views import AirflowBaseView
+from databricks_cli.sdk import JobsService
+from databricks_cli.sdk.api_client import ApiClient
+from flask import flash, redirect, request
+from flask_appbuilder.api import expose
+
+
+def _get_databricks_task_id(task: BaseOperator) -> str:
+    """Get the databricks task ID using dag_id and task_id. removes illegal characters.
+
+    :param task: The task to get the databricks task ID for.
+    :return: The databricks task ID.
+    """
+    return task.dag_id + "__" + task.task_id.replace(".", "__")
+
+
+def get_databricks_task_ids(
+    group_id: str, task_map: dict[str, BaseOperator]
+) -> list[str]:
+    """
+    Returns a list of all Databricks task IDs for a dictionary of Airflow tasks.
+
+    :param group_id: The task group ID.
+    :param task_map: A dictionary mapping task IDs to BaseOperator instances.
+    :return: A list of Databricks task IDs for the given task group.
+    """
+    task_ids = []
+    print("Getting task ids for db")
+    for task_id, task in task_map.items():
+        if task_id != f"{group_id}.launch":
```

Comment:
Will all the needed tasks have this same task ID `f"{group_id}.launch"`?

Reply:
Any databricks job will have a launch task, we don't want to relaunch so we don't add it to the repair.

Reply:
The idea is _not_ to add the group_id.launch task. That's not a databricks task so we shouldn't submit it.

Reply:
@pankajkoti we don't need to worry about that as we are only looking at tasks within that task group.